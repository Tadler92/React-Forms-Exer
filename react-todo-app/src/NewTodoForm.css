.NewTodoForm input {
  margin: 5px 15px 10px 5px;
  padding: 10px;
}

.NewTodoForm button {
  background-color: yellowgreen;
  border-radius: 40px;
}