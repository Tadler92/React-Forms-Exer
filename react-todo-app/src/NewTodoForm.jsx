import { useState } from "react";
import './NewTodoForm.css';

const NewTodoForm = ({addTodo}) => {
  const INITIAL_STATE = {
    taskTodo: "",
  }
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const {name, value} = e.target;
    setFormData(formData => ({
      ...formData,
      [name]: value
    }))
  }

  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({...formData});
    setFormData(INITIAL_STATE);
  }

  return (
    <form className="NewTodoForm" onSubmit={handleSubmit}>
      <label htmlFor="taskTodo">Task Todo: </label>
      <input 
        id="taskTodo"
        type="text" 
        name="taskTodo" 
        placeholder="Task" 
        value={formData.taskTodo}
        onChange={handleChange} 
      />

      <button>Add Todo</button>
    </form>
  )
}



export default NewTodoForm;