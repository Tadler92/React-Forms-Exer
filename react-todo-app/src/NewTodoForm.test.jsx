import {render, fireEvent, waitFor} from "@testing-library/react";
import NewTodoForm from "./NewTodoForm";
import { expect, it } from "vitest";


describe("NewTodoForm Component", () => {
  it("renders without crashing", function () {
    render(<NewTodoForm />);
  });
  
  
  it("matches snapshot", function () {
    const {asFragment} = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});