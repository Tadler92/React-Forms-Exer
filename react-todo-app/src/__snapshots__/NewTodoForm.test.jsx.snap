// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NewTodoForm Component > matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewTodoForm"
  >
    <label
      for="taskTodo"
    >
      Task Todo: 
    </label>
    <input
      id="taskTodo"
      name="taskTodo"
      placeholder="Task"
      type="text"
      value=""
    />
    <button>
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
